### ISSUE TITLE
Regression: Merge command breaks in 0.18

### ISSUE THREAD

**Issue by brookewenig** [databricks/koalas#870]
When I try to run this code in Koalas 0.17, it works fine, but in 0.18, I get this error:
org.apache.spark.sql.AnalysisException: Reference &#39;year&#39; is ambiguous, could be: left_table.year, right_table.year.;

```python
import databricks.koalas as ks
import random


kdf_1 = ks.DataFrame({
    'col_1': range(2010,2020),
    'col_2': [1]*10,
    'col_3': [random.randint(1,100) for x in range(10)]
  
})
kdf_1.columns = ["year", "month", "metric_1"]

kdf_2 = ks.DataFrame({
    'col_1': range(2010,2020),
    'col_2': [1]*10,
    'col_3': [random.randint(1,100) for x in range(10)]
  
})
kdf_2.columns = ["year", "month", "metric_2"]

kdf_1.merge(kdf_2, on=['year', 'month'])
```

**Comment by ueshin** [comment]
@brookewenig Thanks for the report.
It should be fixed in the current master. I tried it with master and it worked.

```py
>>> kdf_1.merge(kdf_2, on=['year', 'month']).sort_values(by=['year', 'month'])
   year  month  metric_1  metric_2
6  2010      1        55        81
0  2011      1        56        78
7  2012      1        50        49
3  2013      1         6        76
4  2014      1        32        11
8  2015      1         5        76
9  2016      1         4        68
2  2017      1        80        65
5  2018      1         3        54
1  2019      1        43        93
```

It will be available in the next release. We are planning to release tonight, or within this week.

**Comment by ueshin** [comment]
I'd close this since this was fixed in `0.19.0`.
